packages/libs/eda/src/lib/map/analysis/mapTypes/plugins/BubbleMarkerMapType.tsx

```diff
@@ -245,6 +247,42 @@ function BubbleMapLayer(props: MapTypeMapLayerProps) {
     [handleSelectedMarkerSnackbars, props.configuration, updateConfiguration]
   );
 
+  // set useEffect for area selection to change selectedMarkers via setSelectedmarkers
+  // define useEffect here to avoid conditional call
+  // thus, this contains duplicate codes, e.g., markers and selectedMarkers
+  useEffect(() => {
+    if (!markersData.error && !markersData.isFetching) {
+      // convert marker data into markers
+      const markers = markersData.data?.markersData?.map((markerProps) => (
```

Comment:
Had a quick look (not tested yet!) and `markersData` seems missing from the dependencies and there's no need to convert `markersData` into `markers`. You can do the bounds check below with `markerData` instead of `markers` (so it's `data.position.lat` instead of `marker.props.position.lat`.

Reply:
@bobular Thank you for your feedbacks 👍 As for markersData-related one, yes it can be simplified so I made a new commit to address them. For the dependency, I checked that markersData-related one causes infinite loop so I didn't add it to the dependency array.